### ISSUE TITLE
Webhook ips whitelist doesn't work

### ISSUE THREAD

**Issue by lucalamo** [n8n-io/n8n#11175]
### Bug Description

In webhook node IPs whitelist doesn't work:
I've setted a comma separated list ex. 1.1.1.1,2.2.2.2 then I call the webhook with a ip client that I've write in the field.
The response code is 403 
The response body: IP is not whitelisted to access the webhook!
In nginx access log:
_access.log:xx.xxx.xx.xxx - - [09/Oct/2024:05:15:31 +0200] "GET /webhook/xxxxxx HTTP/1.1" 403 44 "-" "okhttp/4.2.1"_

Is this right or this field exepects a subnet like: 1.1.1.1/32?

Thank you

### To Reproduce

1. Create a webhook node
2. Set comma separated list with ip likes 1.1.1.1,2.2.2.2 (one have to be your client IP)
3. Call the webhook with your browser 
4. Webhook will execute

### Expected behavior

IPs whitelisting works permitting only ips in comma separated list

### Operating System

ubuntu 22.04

### n8n Version

1.60.1

### Node.js Version

v18.17.1

### Database

PostgreSQL

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @lucalamo,

We have created an internal ticket to look into this which we will be tracking as "GHC-298"

**Comment by Joffcom** [comment]
Hey @lucalamo,

I have just given this a test and using `1.1.1.1,a.b.c.d` has worked and allows my IP to connect.

I suspect the issue is likely to be with your proxy and it is not passing through your IP to the webhook node, If you call the webhook node do you see your IP in the list or the IP of your proxy? I would check for `x-real-ip` and `x-forwarded-for`.

**Comment by lucalamo** [comment]
My nginx section configuration is
`location / {
                proxy_set_header        Connection "Upgrade";
                proxy_set_header        Upgrade $http_upgrade;
                proxy_set_header        Host $host;
                proxy_set_header        X-Real-IP $remote_addr;
                proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
                chunked_transfer_encoding off;
                proxy_buffering off;
                proxy_cache off;
                proxy_pass http://127.0.0.1:5678;
        }`

In access log the ip "xx.xxx.xx.xxx" is the same setted in ips whitelisting
`access.log:xx.xxx.xx.xxx - - [09/Oct/2024:05:15:31 +0200] "GET /webhook/xxxxxx HTTP/1.1" 403 44 "-" "okhttp/4.2.1"`
and it is the X-Real-IP $remote_addr;
I think this is correct, don't you?

**Comment by Joffcom** [comment]
Hey @lucalamo,

If you check in n8n against the webhook execution with the option disabled do you see the header correctly set? I have tested against 2 different n8n installations one beind Traefik and one behind nginx and both are working as expected.

**Comment by lucalamo** [comment]
Yes, both the real ip and the forwarded ip is the same in the ip whitelist option 
![image](https://github.com/user-attachments/assets/1efb61b7-2962-4a96-ba15-c1f0eac3d61e)

and is the same in nginx access_log (the first line is with flag uncheck and the second one with flag checked)
`access.log:xx.xx.xx.xx - - [09/Oct/2024:11:09:32 +0200] "GET /webhook/test HTTP/1.1" 200 34 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36"`
`access.log:xx.xx.xx.xx - - [09/Oct/2024:11:10:11 +0200] "GET /webhook/test HTTP/1.1" 403 44 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36"`

**Comment by Joffcom** [comment]
Interesting, This is going to be tricky as I am not able to reproduce this. If you bypass the proxy and connect directly does that work? That will give us an idea of it is an issue with n8n or an issue in the proxy somewhere.

**Comment by lucalamo** [comment]
I've added $http_x_forwarded_for to nginx access.log as second argument and this is the output
`xx.xx.xx.xx - -  - - [09/Oct/2024:12:57:18 +0200] "GET /webhook/test HTTP/1.1" 403 44 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36"`

It is empty so the directive
` proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;`
is empty.

But why the x-forwarded-for  in webhook node isn't empty?
![image](https://github.com/user-attachments/assets/3d0468c4-4e0e-4b98-9288-1a0a6d8328bf)

**Comment by Joffcom** [comment]
@lucalamo good question I am not sure, We take the value from the request so assuming things are looking good we should show it and use it.

Did you try bypassing the proxy to see if that works?

**Comment by Joffcom** [comment]
I have been able to reproduce this now, Will see if I can work out what is going on.

**Comment by lucalamo** [comment]
Hello,
i didn't try to bypass the proxy because i'm in production env.

Il giorno gio 10 ott 2024 alle ore 17:07 Jon ***@***.***> ha
scritto:

**Comment by Joffcom** [comment]
Hey @lucalamo,

Assuming you only have the one proxy can you set `N8N_PROXY_HOPS` to `1` as an env option and let me know if that helps.

**Comment by lucalamo** [comment]
Hi, it seems to be working now.
Thank you

Il giorno gio 10 ott 2024 alle ore 17:38 Jon ***@***.***> ha
scritto:

**Comment by Joffcom** [comment]
That is perfect thanks for letting me know @lucalamo, We have updated the docs and the node to mention this which should help others and myself in the future.
